src/main/java/liquibase/ext/bigquery/database/BigQueryDatabase.java

```diff
@@ -108,7 +108,7 @@ public boolean isCorrectDatabaseImplementation(DatabaseConnection conn) throws D
     }
     @Override
     public String escapeObjectName(String objectName, Class<? extends DatabaseObject> objectType) {
-        if (objectType.equals(Schema.class) || objectType.equals(Catalog.class)) {
```

Comment:
catalog (ProjectId) can contain dashes, so we can't skip escaping it with

Reply:
Apart the refactoring, was it the fix for DAT-18728? Or did you do anything else to fix it?

Reply:
Main fix was for building and executing query in `BigQueryForeignKeySnapshotGenerator.addTo`
```
String query = String.format("SELECT CONSTRAINT_NAME FROM %s.%s.TABLE_CONSTRAINTS WHERE TABLE_NAME=?
AND TABLE_SCHEMA=? AND TABLE_CATALOG=? AND " +
"CONSTRAINT_TYPE='FOREIGN KEY';", jdbcSchemaName, database.getSystemSchema().toUpperCase());
```
This query executed by `Executor.queryForList` with adding table, catalog and schema names was invalid. 
BigQuery doesn't allow dashes in the schema, but it allows them in the catalog (ProjectID). Moreover, random names generated for ProjectID often have dashes. 
So I moved the catalog name to the object identifier (catalogName.schemaName.INFORMATION_SCHEMA.TABLE_CONSTRAINTS) and escaped it with backquotes using updated `escapeObjectName()` method

apart from that i also fixed a few places where we had `schema.getSchema()` instead of `schema.getName()` when building query

Reply:
Thanks!